### ISSUE TITLE
Unstructured solid mesh

### ISSUE THREAD

**Issue by VamosZhi** [solids4foam/solids4foam#73]
Could solids4foam be applied to the solid domain with unstructured mesh?
I test one simple case, the structured mesh works, and the unstructured mesh crashes.
Sometimes it's really difficult to create the structured mesh for the geometrics of a solid domain.

Thanks!

**Comment by philipcardiff** [comment]
Yes, solids4foam and OpenFOAM work with unstructured meshes. But, of course, some discretisation may be less stable on some types of meshes.

If you have a small example case, please upload it here.

**Comment by Youngxl1998** [comment]
Oh, I am sorry about that, thanks for your suggestion.

**Comment by VamosZhi** [comment]
Hi, Philip,

Thanks for your response! What I mean is that it seems unstructured mesh is not supported by the solid domain. My case file is attached. The solid domain here uses unstructured mesh (all Tets cell type) like:
![image](https://github.com/solids4foam/solids4foam/assets/73256102/1ca2f5af-a89f-421b-865f-38d87602ae1f)

But the case crashed after several time steps as:
![image](https://github.com/solids4foam/solids4foam/assets/73256102/93843df9-707f-42d1-afd9-00780d5116e5)

While using structured mesh for solid domain, the case runs smoothly. 
![image](https://github.com/solids4foam/solids4foam/assets/73256102/be272c7a-3e72-44b5-85cb-52467c88400d)

I need to use unstructured mesh for the solid domain because the concerned geometrics will be complicated in the future. Thanks!

Zhi

**Comment by VamosZhi** [comment]
CASE file is attached as google drive link here:
https://drive.google.com/file/d/1YbAALb91pRb7PMbiaMOLuDY7P7iX2dI7/view?usp=sharing

**Comment by philipcardiff** [comment]
If the same issue can be reproduced with a recent solids4Foam/OpenFOAM version, please open a new issue or ask for this one to be reopened with a minimal current test case.

For unstructured tetrahedral solid grids, the newer higher-order solid solver is likely the better route. A useful starting point is the sphericalCavity tutorial, which includes a high-order option:

```bash
cd tutorials/solids/linearElasticity/sphericalCavity
./Allrun highOrder
```
